package ConfigTask;

import "CommonGiftConfig.proto";


message TaskCondition
{
	repeated uint32   value     = 1;
}
message TaskCPP
{
	required   uint32  id                                        = 1;//任务id
	required   uint32  pagetype                                  = 2;//分页类型
	required   uint32  tasktype                                  = 3;//任务类型
	repeated   TaskCondition  condition                          = 4;//任务条件
	repeated   uint32  priority                                  = 5;//优先级
	repeated   CommonGiftConfig.CommonModifyItem   reward        = 6;//任务奖励	
}

message MissionCPP
{
	required uint32 id              = 1;//当前任务
	repeated uint32 next            = 2;//下一任务id
	required uint32 type            = 3;//任务类型
	required uint32 statisticalType = 4;//任务类型
	repeated uint32 condition       = 5;//任务条件
	required   CommonGiftConfig.CommonModifyItem   reward        = 6;//任务奖励	
}


message TaskInfo
{
	repeated TaskCPP   task                             = 1;
	repeated MissionCPP missions                        = 2;//单线任务配置
}